pkg/logger/sugared.go

```diff
@@ -39,17 +56,66 @@ func (s *sugared) ErrorIfFn(fn func() error, msg string) {
 	}
 }
 
-// AssumptionViolation wraps Error logs with assumption violation tag.
+const assumptionViolationPrefix = "AssumptionViolation: "
+
 func (s *sugared) AssumptionViolation(args ...interface{}) {
-	s.h.Error(append([]interface{}{"AssumptionViolation:"}, args...))
+	s.h.Error(append([]interface{}{assumptionViolationPrefix}, args...))
 }
 
-// AssumptionViolationf wraps Errorf logs with assumption violation tag.
 func (s *sugared) AssumptionViolationf(format string, vals ...interface{}) {
-	s.h.Errorf("AssumptionViolation: "+format, vals...)
+	s.h.Errorf(assumptionViolationPrefix+format, vals...)
 }
 
-// AssumptionViolationw wraps Errorw logs with assumption violation tag.
 func (s *sugared) AssumptionViolationw(msg string, keyvals ...interface{}) {
-	s.h.Errorw("AssumptionViolation: "+msg, keyvals...)
+	s.h.Errorw(assumptionViolationPrefix+msg, keyvals...)
+}
+
+const critPrefix = "[crit] "
+
+func (s *sugared) Critical(args ...interface{}) {
+	switch t := s.h.(type) {
+	case *logger:
```

Comment:
The base Logger in this repo does not have `Critical`. Only the core repo Logger has those extra methods.

Reply:
I don't follow, but I approved because I think this is outside the scope of this PR and can be addressed in a followup if needed. 

These switch cases first look for a match if `s.h` is of type `*logger`. If it's not, they look whether s.h has implemented `Critical`. The case where the logger is not of the type *logger and does not have `Critical` results in call to `s.h.Error(append([]any{critPrefix}, args...)...)`

I'm curious about the first statement - when looking for whether `s.h` is of type `*logger`

Reply:
There are test cases here: https://github.com/smartcontractkit/chainlink-common/blob/sugared-logger/pkg/logger/logger_test.go#L142-L153